pyproject.toml

```diff
@@ -70,10 +70,11 @@ scikit-learn = "^1.2.1"
 pytest-cov = "^4.0.0"
 pylint = "3.1.0"
 matplotlib = "^3.7.0"
-tensorflow = { version = "^2.16.1", markers = "sys_platform == 'linux'" }
+tensorflow = { version = "^2.16.1", markers = "sys_platform == 'linux' or sys_platform == 'darwin'", optional = true }
```

Comment:
```suggestion
```
why do we need explicit dependence from `tensorflow`? this should be taken care of by the `qiboml` dependency, as it happens for `qibojit` and `numba`.

Reply:
I am keeping this because many tests are importing tensorflow (just to give an example: [this one](https://github.com/qiboteam/qibo/blob/958b11bc38b800f217ad9c97b950635896935fd4/tests/test_backends_torch_gradients.py#L50)). 

I agree many of the imports can be removed, but probably this is something we can solve after the migration.

Reply:
So, now we can get safely rid of this dependency, right?